### ISSUE TITLE
Batch-Cluster rejects Promises with a non-Error

### ISSUE THREAD

**Issue by nknapp** [mceachen/batch-cluster.js#3]
I noticed the following error message when using `exiftool-vendored` in my mocha-tests:

```
Error: the string "stderr.data: Warning: No writable tags set from tmp/ffmpeg/lumix/PRIVATE/AVCHD/BDMV/STREAM/00000.MTS" was thrown, throw an Error :)
```

I think, the reject function of a promise should always be called with an Error, not a string. One particular reason is, that stack-traces cannot be transported on a string.

[This line in BatchProcess.ts](https://github.com/mceachen/batch-cluster.js/blob/25170776fa3e631b841b2cb8d82d1aa8c92c2ddd/src/BatchProcess.ts#L252) calls the "onError"-function with a string as parameter. [This line in Task.ts](https://github.com/mceachen/batch-cluster.js/blob/d2de9bc28fa8fb864f675b6a06d0f3528a0d0cad/src/Task.ts#L66) calls the "reject" function of the deferred with this string, which ultimately leads to the Promise being rejected with a string as "reason". I think this should be changed, although I can see this is a breaking change.

I'd be happy to create a PR, if you like.

**Comment by mceachen** [comment]
Hey, sorry for the delay, this didn't show in my github notifications!
You are right--I think I was concerned about the weight of Error creation, but that's certainly premature optimization in retrospect. I'll update now.

**Comment by mceachen** [comment]
Released both v3.0.0 batch-cluster and v5.0.0 of exiftool-vendored. Thanks again for the idea.
